packages/core/src/tools.ts

```diff
@@ -269,6 +274,47 @@ export namespace Tools {
 		}
 	}
 
+	/**
+	 * Update legend additional items
+	 * @param {any} defaultOptions
+	 * @param {any} providedOptions
+	 */
+	export function updateLegendAdditionalItems(
+		defaultOptions,
+		providedOptions
+	) {
+		let defaultLegendItems = Tools.getProperty(
+			defaultOptions,
+			'legend',
+			'additionalItems'
+		);
+		const providedLegendItems = Tools.getProperty(
+			providedOptions,
+			'legend',
+			'additionalItems'
+		);
+
+		// Retain default legend additional items
+		if (defaultLegendItems && providedLegendItems) {
+			const providedTypes = providedLegendItems.map((item) => item.type);
+
+			// Remove same type default items if they are provided
+			defaultLegendItems = defaultLegendItems.filter(
+				(item) =>
+					!providedTypes.includes(item.type) &&
+					item.type === LegendItemType.RADIUS
+			);
+
+			providedOptions.legend.additionalItems = Tools.unionBy(
+				defaultLegendItems,
+				providedLegendItems,
+				'name'
+			);
+
+			defaultOptions.legend.additionalItems = [];
```

Comment:
I think it'd be more readable to call them user-provided rather than just provided